java/ql/test/experimental/query-tests/security/CWE-522-DecompressionBombs/CommonsCompressHandler.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.commons.compress.archivers.cpio.CpioArchiveInputStream;
+import org.apache.commons.compress.archivers.jar.JarArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.compressors.lzma.LZMACompressorInputStream;
+import org.apache.commons.compress.compressors.pack200.Pack200CompressorInputStream;
+import org.apache.commons.compress.compressors.snappy.SnappyCompressorInputStream;
+import org.apache.commons.compress.compressors.xz.XZCompressorInputStream;
+import org.apache.commons.compress.compressors.zstandard.ZstdCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.brotli.BrotliCompressorInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.deflate.DeflateCompressorInputStream;
+import org.apache.commons.compress.compressors.deflate64.Deflate64CompressorInputStream;
+import org.apache.commons.compress.compressors.z.ZCompressorInputStream;
+
+public class CommonsCompressHandler {
+
+    static void commonsCompressArchiveInputStream(InputStream inputStream) throws ArchiveException {
+        new ArArchiveInputStream(inputStream); // $ hasTaintFlow="inputStream"
+        new ArjArchiveInputStream(inputStream); // $ hasTaintFlow="inputStream"
+        new CpioArchiveInputStream(inputStream); // $ hasTaintFlow="inputStream"
+        new JarArchiveInputStream(inputStream); // $ hasTaintFlow="inputStream"
+        new ZipArchiveInputStream(inputStream); // $ hasTaintFlow="inputStream"
+    }
+
+    public static void commonsCompressorInputStream(InputStream inputStream) throws IOException {
+        BufferedInputStream in = new BufferedInputStream(inputStream);
+        OutputStream out = Files.newOutputStream(Path.of("tmpfile"));
+        GzipCompressorInputStream gzIn = new GzipCompressorInputStream(in); // $ hasTaintFlow="in"
+        // for testing
+        new BrotliCompressorInputStream(in); // $ hasTaintFlow="in"
+        new BZip2CompressorInputStream(in); // $ hasTaintFlow="in"
+        new DeflateCompressorInputStream(in); // $ hasTaintFlow="in"
+        new Deflate64CompressorInputStream(in); // $ hasTaintFlow="in"
+        new BlockLZ4CompressorInputStream(in); // $ hasTaintFlow="in"
+        new LZMACompressorInputStream(in); // $ hasTaintFlow="in"
+        new Pack200CompressorInputStream(in); // $ hasTaintFlow="in"
+        new SnappyCompressorInputStream(in); // $ hasTaintFlow="in"
+        new XZCompressorInputStream(in); // $ hasTaintFlow="in"
+        new ZCompressorInputStream(in); // $ hasTaintFlow="in"
+        new ZstdCompressorInputStream(in); // $ hasTaintFlow="in"
+
+        int buffersize = 4096;
+        final byte[] buffer = new byte[buffersize];
+        int n = 0;
+        while (-1 != (n = gzIn.read(buffer))) {
```

Comment:
Should `gzIn` be a sink here? If yes, why is there no test result here? (Maybe because `buffer` is not tainted, but you could change that.) If no, why is this code here in the test?

Please go through all the tests making sure that they are testing what you want them to test and that they are getting results when you want them to.